app/backend/routes/user/profiles.ts

```diff
@@ -0,0 +1,78 @@
+import { Hono } from "hono";
+import { zValidator } from "@hono/zod-validator";
+import { HTTPException } from "hono/http-exception";
+import { eq, sql } from "drizzle-orm";
+import db from "../../db/db";
+import { userProfileTable } from "../../db/schema.sql";
+import { profileEditSchema, GUIDParamSchema } from "./types";
+import { protectedRoute, limiter } from "../../lib/middleware";
+import type { User, Session } from "#db/schema.sql";
+
+//* Define the profile route
+
+const profileRoute = new Hono<{
+	Variables: { user: User; session: Session };
+}>();
```

Comment:
Use factory function to create hono router instances.

```suggestion
import { createRouter } from "#/lib/factory"

const profileRouter = createRouter();
```

Reply:
Ah didnt know, i changed it.